### PR TITLE
Add a method chain and callback standard for Javascript

### PR DESCRIPTION
**Pull request by DanielBrookRoberge**
Turns out I have *opinions* on formatting method chains/fluent APIs in Javascript. I propose them here as an addition to our style guide. Discuss!

Status: **Ready for review**

Reviewer: @jansepar and the rest of the JS devs.

### CODE REVIEW

**Comment by johnboxall** [comment]
By analyzing the blame information on this pull request, we identified @scalvert, @ry5n and @tedtate to be potential reviewers.

**Comment by ericmuyser** [comment]
Congrats guys, johnbot picks you! :laughing:

File: javascript/README.md [hunk 1]
```diff
@@ -402,3 +402,61 @@ Prefer the short, more succinct version.
...
+The break between two function expressions passed as arguments can be
+quite awkward. It is clearer to use named functions, or to change your
+use of the API to split them up.
+
+```javascript
+// Bad
+functionThatReturnsAPromise()
+    .then(function() {
+        // code here!
+    }), function() {
```
Comment (tedtate) [inline javascript/README.md]: `}, function() {`

**Comment by scalvert** [comment]
hawt

File: javascript/README.md [hunk 2]
```diff
@@ -402,3 +402,61 @@ Prefer the short, more succinct version.
...
+
+###Place each call on its own line, beginning with the period, and indented
+
+The sequence of operations is clearer if each 'step' in the process
+begins at the start of the line. Indenting relative to the previous
+line makes the chain distinct.
+
+```javascript
+// Good
+return functionThatReturnsAPromise()
```
Comment (scalvert) [inline javascript/README.md]: +1!

**Comment by scalvert** [comment]
:+1: 

LGTM!

**Comment by jansepar** [comment]
:+1:

**Comment by DanielBrookRoberge** [comment]
Thanks everyone for your attention!

**Pull request merged**